app/src/app/components/header/header.component.html

```diff
@@ -19,6 +19,7 @@ <h1><a routerLink="/">ATT&CK Workbench&nbsp;<span class="app-version lowercase">
     <button mat-button *ngIf="!isLoggedIn else account" [ngStyle]="{'visibility': showHamburger ? 'hidden' : 'visible'}" (click)="login()">
         login
     </button>
+    <button mat-button *ngIf="!isLoggedIn" [ngStyle]="{'visibility': showHamburger ? 'hidden' : 'visible'}" (click)="register()">register</button>
```

Comment:
The register button in the header should only appear if the authentication mechanism is configured to `oidc`, since `anonymous` users don't need to register to use the application. The button doesn't do anything when clicked because of a later check for the mechanism type in the call stack, but it should be hidden when configured to `anonymous`.